test/Interop/Cxx/operators/member-inline-silgen.swift

```diff
@@ -32,3 +32,54 @@ public func call(_ wrapper: inout AddressOnlyIntWrapper) -> Int32 { wrapper() }
 // CHECK: end_access [[SELFACCESS]] : $*AddressOnlyIntWrapper
 
 // CHECK: sil [clang AddressOnlyIntWrapper.callAsFunction] [[NAME]] : $@convention(c) (@inout AddressOnlyIntWrapper) -> Int32
+
+public func index(_ arr: inout ReadOnlyIntArray, _ arg: Int32) -> Int32 { arr[arg] }
+
+// CHECK: sil @[[FUNCNAME:.*]] : $@convention(thin) (@inout ReadOnlyIntArray, Int32) -> Int32 {
+// CHECK: bb0([[SELF:%.*]] : $*ReadOnlyIntArray, [[INDEX:%.*]] : $Int32):
+// CHECK:   [[SELFACCESS:%.*]] = begin_access [modify] [static] [[SELF]] : $*ReadOnlyIntArray
+// CHECK:   [[SELFACCESS2:%.*]] = begin_access [modify] [static] [[SELFACCESS]] : $*ReadOnlyIntArray
+// CHECK:   [[OP:%.*]] = function_ref [[READCLASSNAME:@(_ZNK16ReadOnlyIntArrayixEi|\?\?AReadOnlyIntArray@@QEBAAEBHH@Z)]] : $@convention(c) (@inout ReadOnlyIntArray, Int32) -> UnsafePointer<Int32>
+// CHECK:   [[PTR:%.*]] = apply [[OP]]([[SELFACCESS2]], [[INDEX]]) : $@convention(c) (@inout ReadOnlyIntArray, Int32) -> UnsafePointer<Int32>
+// CHECK:   end_access [[SELFACCESS2]] : $*ReadOnlyIntArray
+// CHECK:   [[PTR2:%.*]] = struct_extract [[PTR]] : $UnsafePointer<Int32>, #UnsafePointer._rawValue
+// CHECK:   pointer_to_address [[PTR2]] : $Builtin.RawPointer to [strict] $*Int32
+// CHECK:   end_access [[SELFACCESS]] : $*ReadOnlyIntArray
+// CHECK: } // end sil function '[[FUNCNAME]]'
+
+// CHECK: sil shared [transparent] [serializable] @[[FUNCNAME:.*]] : $@convention(method) (Int32, @inout ReadOnlyIntArray) -> Int32 {
+// CHECK: bb0([[INDEX:%.*]] : $Int32, [[SELF:%.*]] : $*ReadOnlyIntArray):
+// CHECK:   [[SELFACCESS:%.*]] = begin_access [modify] [static] [[SELF]] : $*ReadOnlyIntArray
+// CHECK:   [[OP:%.*]] = function_ref [[READCLASSNAME]] : $@convention(c) (@inout ReadOnlyIntArray, Int32) -> UnsafePointer<Int32>
+// CHECK:   [[PTR:%.*]] = apply [[OP]]([[SELFACCESS]], [[INDEX]]) : $@convention(c) (@inout ReadOnlyIntArray, Int32) -> UnsafePointer<Int32>
+// CHECK:   end_access [[SELFACCESS]] : $*ReadOnlyIntArray
+// CHECK:   [[PTR2:%.*]] = struct_extract [[PTR]] : $UnsafePointer<Int32>, #UnsafePointer._rawValue
+// CHECK:   pointer_to_address [[PTR2]] : $Builtin.RawPointer to [strict] $*Int32
+// CHECK: } // end sil function '[[FUNCNAME]]'
+
+public func index(_ arr: inout ReadWriteIntArray, _ arg: Int32, _ val: Int32) { arr[arg] = val }
+
+// CHECK: sil @[[FUNCNAME:.*]] : $@convention(thin) (@inout ReadWriteIntArray, Int32, Int32) -> () {
```

Comment:
You don't need a regex expression here, I think we actually want the full function name. That shouldn't ever change, so we don't really get anything from the pattern matching. And `FUNCNAME` isn't really that helpful for those reading the tests. 

I'm OK with leaving `READWRITECLASSNAME` because a) that one actually says what it is, and b) it is set to a specific function name, `_ZN17ReadWriteIntArrayixEi|\?\?AReadWriteIntArray@@QEAAAEAHH@Z`.

Reply:
Okay, I've replaced `FUNCNAME` with the actual function name.